evalml/pipelines/pipeline_base.py

```diff
@@ -425,6 +426,58 @@ def feature_importance(self):
         df = pd.DataFrame(importance, columns=["feature", "importance"])
         return df
 
+    def graph_json(self):
+        """Generates a JSON with nodes consisting of the component names and parameters, and edges detailing component relationships.
+
+        This can be used to build graphs across a variety of tools.
+        Template: {"Edges": {"from_parent_node_to_child_node" : {"data": what_value_is_passed,
+                                                                 "from": "parent_node",
+                                                                 "to": "child_node"},
+                            ...},
+                    "Nodes": {"component_name": {"class": class_name,
+                                                 "attributes": parameters_attributes},
+                            ...}
+                    }
+
+        Returns:
+            dag_json (str): A serialized JSON representation of a DAG structure.
+        """
+        dag_json = {"Nodes": {"X": "X", "y": "y"}, "Edges": {}}
+
+        for (
+            component_name,
+            component_info,
+        ) in self.component_graph.component_dict.items():
+            dag_json["Nodes"][component_name] = {
+                "class": component_info[0]
+                if isinstance(component_info[0], str)
+                else component_info[0].name,
+                "attributes": self.parameters[component_name]
+                if component_name in self.parameters
+                else None,
+            }
+
+        for (
+            component_name,
+            component_info,
+        ) in self.component_graph.component_dict.items():
+            for input_ in component_info[1:]:
+                from_comp = input_[:-2]
+                if input_.endswith(".x") or input_.endswith(".y"):
+                    dag_json["Edges"][f"from_{from_comp}_to_{component_name}"] = {
```

Comment:
Curious, what do you think about (`from_{from_comp}_to_{component_name}`)? Do we need it? It seems like something that the users could construct from the to / from, and feels a little more cluttered to me 🤔

Reply:
Agreed.  I think @freddyaboulton 's proposal for an alternate structure is a little leaner.  I think we could do away with that key and just have the edges be either a list of tuples where the meaning of the data is positional and documented, e.g. [(from, to), (from, to)...] or, if you want to retain some of your explicit structure, a list of dict: [{"from": from, "to": to, "data":data}...]